packages/services/src/validators/arena-config.validator.ts

```diff
@@ -0,0 +1,247 @@
+import { z } from "zod/v4";
+
+import {
+  CrossChainTradingTypeSchema,
+  EvaluationMetricSchema,
+  TradingConstraintsSchema,
+} from "../types/index.js";
+
+/**
+ * Arena Config Validators
+ * Based on: https://www.notion.so/Arena-Config-v1-Fields-Types-Purpose-294dfc9427de8088a0bad74538624ae7
+ */
+
+/**
+ * Base arena config schema
+ */
+export const ArenaConfigSchema = z
+  .object({
+    apiVersion: z.literal("arenas.recall/v1"),
+    kind: z.literal("Competition"),
+    metadata: z.object({
+      id: z
+        .string()
+        .regex(
+          /^[a-z0-9-]+$/,
+          "ID must be lowercase kebab-case (e.g., 'spot-sharpe-aero-2025w44')",
+        ),
+      name: z.string().min(4).max(80),
+      createdBy: z.string(),
+    }),
+    classification: z.object({
+      category: z.enum(["crypto_trading"]),
+      skill: z.string(),
+      venues: z.array(z.string()).optional(),
+      chains: z.array(z.string()).optional(),
+    }),
+    schedule: z
+      .object({
+        registration: z.object({
+          opensAt: z.coerce.date(),
+          closesAt: z.coerce.date(),
+        }),
+        runs: z.object({
+          startAt: z.coerce.date(),
+          endAt: z.coerce.date(),
+          lateJoinPolicy: z
+            .enum(["from_join_to_end", "entire_window"])
+            .optional(),
+        }),
+        boosting: z
+          .object({
+            startAt: z.coerce.date(),
+            endAt: z.coerce.date(),
+          })
+          .optional(),
+      })
+      .optional(),
+    participation: z.object({
+      maxAgents: z.number().int().min(1).optional(),
+      vips: z.array(z.string()).optional(),
+      allowlist: z.array(z.string()).optional(),
+      blocklist: z.array(z.string()).optional(),
+      requirements: z
+        .object({
+          minStake: z.number().min(0).optional(),
+          recallRank: z.number().optional(),
+          allowlistOnly: z.boolean().optional(),
+        })
+        .optional(),
+    }),
+    engine: z.object({
+      id: z.string(),
+      version: z.string().regex(/^\d+\.\d+\.\d+$/, "Must be valid semver"),
+      description: z.string().optional(),
+      details: z.record(z.string(), z.unknown()).optional(),
+      params: z.record(z.string(), z.unknown()).optional(),
```

Comment:
Could be more specific here and leverage discriminated unions https://zod.dev/api?id=discriminated-unions